azurerm/resource_arm_storage_management_policy_test.go

```diff
@@ -0,0 +1,207 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+)
+
+func TestAccAzureRMStorageManagementPolicy_basic(t *testing.T) {
+	resourceName := "azurerm_storage_management_policy.testpolicy"
+	ri := tf.AccRandTimeInt()
+	rs := acctest.RandString(4)
+	location := testLocation()
+	config := testAccAzureRMStorageManagementPolicy_basic(ri, rs, location)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.name", "rule1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.type", "Lifecycle"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.filters.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.filters.0.prefix_match.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.filters.0.prefix_match.3439697764", "container1/prefix1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.filters.0.blob_types.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.filters.0.blob_types.1068358194", "blockBlob"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.base_blob.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.base_blob.0.tier_to_cool_after_days_since_modification_greater_than", "10"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.base_blob.0.tier_to_archive_after_days_since_modification_greater_than", "50"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.base_blob.0.delete_after_days_since_modification_greater_than", "100"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.snapshot.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "rule.0.actions.0.snapshot.0.delete_after_days_since_creation_greater_than", "30"),
+				),
+			},
```

Comment:
We need to check that the resource is being imported properly. You should add the below just after every test step. That import step also checks that the data is what we expect so you won't have to do all of the `resource.TestCheckResourceAttr` you did above
```suggestion
			},
			{
				ResourceName:      resourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
```